Automattic/jetpack
# AI Assistant: Enhance thumbs feedback on AI Assistant and extensions

Fixes #40734 

## Proposed changes:

* Moves the prompt types to AI Client
* Changes thumbs prompt implementation on extensions away from relying on the placeholder
* Adds a `lastAction` state to the block and the extensions to track the prompt on the thumbs event
* Adds the block type to the event parameters

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor with the thumbs feedback feature enabled
* Use the AI Assistant block and its extensions
* After a request, open the deveoper console and click on a feedback icon
* Check that the event has the block type and the prompt, which is either a user prompt or a mapping of the toolbar action used, like "Translate: Korean"
* Since some data was moved around, check that the AI Assistant and the extensions work as expected.

**Known issue**: The feedback is not displayed when using the list to table feature yet
